docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md

```diff
@@ -0,0 +1,85 @@
+
+With `queryType` populated in queries and sent to your backend plugin below is an example of how you would use the [`datasource.QueryTypeMux`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/datasource#QueryTypeMux) to multiplex or route different query types to separate query handlers.
+
+Implemented in this way, each query handler can then `json.Unmarshal` each query JSON field in [`DataQuery`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend#DataQuery) to a certain Go struct as shown in this example:
+
+```
+package mydatasource
+
+import (
+	"context"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
+)
+
+type MyDatasource struct {
+	queryHandler backend.QueryDataHandler
+}
+
+func New() *MyDatasource {
+	ds := &MyDatasource{}
+	queryTypeMux := datasource.NewQueryTypeMux()
+	queryTypeMux.HandleFunc("metrics", ds.handleMetricsQuery)
+	queryTypeMux.HandleFunc("logs", ds.handleLogsQuery)
+	queryTypeMux.HandleFunc("traces", ds.handleTracesQuery)
+	queryTypeMux.HandleFunc("", ds.handleQueryFallback)
+	ds.queryHandler := queryTypeMux
+	return ds
+}
+
+func (d *MyDatasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	return d.queryHandler.QueryData(ctx, req)
+}
+
+// handleMetricsQuery handle queries of query type "metrics".
+// All queries in backend.QueryDataRequest is guaranteed to only
+// include queries with queryType "metrics".
+func (d *MyDatasource) handleMetricsQuery(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// implementation...
+}
+
+// handleLogsQuery handle queries of query type "logs".
+// All queries in backend.QueryDataRequest is guaranteed to only
+// include queries with queryType "logs".
+func (d *MyDatasource) handleLogsQuery(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// implementation...
+}
+
+// handleTracesQuery handle queries of query type "logs".
+// All queries in backend.QueryDataRequest is guaranteed to only
+// include queries with queryType "traces".
+func (d *MyDatasource) handleTracesQuery(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// implementation...
+}
+
+// handleQueryFallback handle queries without a matching query type handler registered.
+func (d *MyDatasource) handleQueryFallback(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// implementation...
+}
+```
+
+## Advanced usage
+
+An example of using [`QueryTypeMux`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/datasource#QueryTypeMux) can be found for Grafana's built-in Grafana Azure Monitor data source. Refer to this code for examples of implementation:
```

Comment:
We don't cover this in https://github.com/grafana/grafana-plugin-examples yet, but now we have the testdata datasource using it so linking to that.

```suggestion
An example of using [`QueryTypeMux`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/datasource#QueryTypeMux) can be found for Grafana's built-in TestData data source. Refer to this code for examples of implementation:
```